src/js/models/layerstructure.js

```diff
@@ -1604,7 +1604,8 @@ define(function (require, exports, module) {
 
             // .set is used here instead of merge to eliminate the other styles
             var nextText = layer.text.set(styleProperty, nextStyle),
-                nextLayer = layer.set("text", nextText);
+                tmpLayer = layer.set("text", nextText),
+                nextLayer = tmpLayer.set("textWarningLevel", 0);
```

Comment:
modifying layers with missing fonts, ( with a suppressed dialog) should substitute out the missing font. 
at least that is the behavior in classic